service/tbc/rpc_test.go

```diff
@@ -558,57 +492,49 @@ func TestBalanceByAddress(t *testing.T) {
 					t.Fatal(ctx.Err())
 				}
 				indexAll(ctx, t, tbcServer)
-				lastErr = nil
-				err = tbcapi.Write(ctx, tws.conn, "someid", tbcapi.BalanceByAddressRequest{
+
+				if err = tbcapi.Write(ctx, tws.conn, "someid", tbcapi.BalanceByAddressRequest{
 					Address: tti.address(),
-				})
-				if err != nil {
-					lastErr = err
-					continue
+				}); err != nil {
+					t.Fatal(err)
 				}
 
 				var v protocol.Message
-				err = wsjson.Read(ctx, c, &v)
-				if err != nil {
-					lastErr = err
-					continue
+				if err = wsjson.Read(ctx, c, &v); err != nil {
+					t.Fatal(err)
 				}
 
-				if v.Header.Command == tbcapi.CmdBalanceByAddressResponse {
-					if err := json.Unmarshal(v.Payload, &response); err != nil {
-						t.Fatal(err)
-					}
+				if v.Header.Command != tbcapi.CmdBalanceByAddressResponse {
+					t.Fatalf("received unexpected command: %s", v.Header.Command)
+				}
 
-					var pricePerBlock uint64 = 50 * 100000000
-					var blocks uint64 = 4
-					var expectedBalance uint64 = 0
-					if !tti.doNotGenerate {
-						expectedBalance = pricePerBlock * blocks
-					}
+				if err = json.Unmarshal(v.Payload, &response); err != nil {
+					t.Fatal(err)
+				}
 
-					expected := tbcapi.BalanceByAddressResponse{
-						Balance: expectedBalance,
-						Error:   nil,
-					}
-					if diff := deep.Equal(expected, response); len(diff) > 0 {
-						if response.Error != nil {
-							t.Error(response.Error.Message)
-						}
-						t.Logf("unexpected diff: %s", diff)
-
-						// there is a chance we just haven't finished indexing
-						// the blocks and txs, retry until timeout
-						continue
-					}
-					break
-				} else {
-					lastErr = fmt.Errorf("received unexpected command: %s", v.Header.Command)
+				var pricePerBlock uint64 = 50 * 100000000
+				var blocks uint64 = 4
+				var expectedBalance uint64 = 0
+				if !tti.doNotGenerate {
+					expectedBalance = pricePerBlock * blocks
 				}
 
-			}
+				expected := tbcapi.BalanceByAddressResponse{
+					Balance: expectedBalance,
+					Error:   nil,
+				}
+				if diff := deep.Equal(expected, response); len(diff) > 0 {
+					if response.Error != nil {
+						t.Error(response.Error.Message)
+					}
+					t.Logf("unexpected diff: %s", diff)
 
-			if lastErr != nil {
-				t.Fatal(lastErr)
+					// there is a chance we just haven't finished indexing
+					// the blocks and txs, retry until timeout
+					continue
+				} else {
+					break
+				}
```

Comment:
I think this `else` statement can be removed to simplify this logic.
```suggestion
				}

				break
```